### PR TITLE
feature: #365 upgraded timeout flag

### LINKED ISSUE #365: Adding a Hour and Minute  `--timeout` in chaosblade create command
## Issue Description
when initiating  the chaos blade with a time tag like (`--timeout  90`) for 90 seconds it would simpler  to write like `1m30s` or `01:30`
```
blade create cpu load --timeout 1m30s
blade create cpu load --timeout 01:30
```

Type: *feature request*

### Describe  what feature you want
Having CLI option to take an argument in terms of minutes and hours like seconds would become little user friendly.

### Describe what you expected to happen
have this `time` tag a logic to convert these strings to seconds

### PR DESCRIPTION
**Pull request by Swarag-N**
### Describe what this PR does / why we need it
This PR is for issue #365 . This PR converts the timeout flag from a time interval `String` to a `Number`. It would be simpler for developers to give timeout as 2m ( 2 minutes ), 2m30s  than 120 and 90.

### Does this pull request fix one issue?
Fixes #365 

### Describe how you did it
Regex module is used parsing the string and get Number. Two regular expressions are used.

1. To verify whether the time interval is in the correct format. [here](https://www.debuggex.com/r/rHj0N3qflvDVuz_L)
image

```^(\d+h)?(\d+m)?(\d+s)?$```

2. To form sub-strings for a timeout as Number. [here](https://www.debuggex.com/r/CPUCqkHD2JBuooCn)
image

```^(\d+h)?(\d+m)?(\d+s)?$```

### Describe how to verify it
Compile and test with `blade create cpu load --timeout 2m3s`
image

```sw4r4g@Acer-Avenger:/mnt/s/docker/test/chaosblade-0.6.0$ ./blade status a41bc92641c90a88
{
        "code": 200,
        "success": true,
        "result": {
                "Uid": "a41bc92641c90a88",
                "Command": "cpu",
                "SubCommand": "fullload",
                "Flag": " --timeout=1m --cpu-percent=20",
                "Status": "Destroyed",
                "Error": "",
                "CreateTime": "2020-06-24T19:41:24.480988+05:30",
                "UpdateTime": "2020-06-24T19:42:31.5633272+05:30"
        }
}
```

### CODE REVIEW

**Comment by xcaspar** [comment]
Why don't you use time.ParseDuration func?

**Comment by Swarag-N** [comment]
I was not familiar with `time.ParseDuration` function.  This function does provide better error handling than the regex version. I shall change it.
Thank You for giving a suggestion.

**Comment by Swarag-N** [comment]
I have refactored the code to use `time.ParseDuration`. The old way of setting flag is not disturbed (Passing Number).
Error is thrown when the flag is not either a `number` or `time duration`.

```
root@f939a9139919:/app# ls
DockerFile  bin  blade  lib
root@f939a9139919:/app# ./blade create cpu load --timeout 23w //fails
time: unknown unit w in duration 23w
root@f939a9139919:/app# ./blade create cpu load --timeout 23 //passes
{"code":200,"success":true,"result":"e81105504dfa17f1"}
root@f939a9139919:/app# ./blade destroy e81105504dfa17f1
{"code":200,"success":true,"result":"command: cpu fullload  --timeout=23, destroy time: 2020-06-28T11:22:09.5419411Z"}
root@f939a9139919:/app# ./blade create cpu load --timeout 1m2s //passes
{"code":200,"success":true,"result":"21291b73661c421d"}
root@f939a9139919:/app# ./blade destroy 21291b73661c421d
{"code":200,"success":true,"result":{"Target":"cpu","Scope":"","ActionName":"fullload","ActionFlags":{"timeout":"1m2s"}}}
```
### Describe how you did it
Used `time.ParseDuration` to raise an error (if) and passes the duration as Seconds.

<details>
  <summary>I have run it locally</summary>
```
root@f939a9139919:/app# ./blade s --type c
{
        "code": 200,
        "success": true,
        "result": [
                {
                        "Uid": "41aa902b4a74d201",
                        "Command": "cpu",
                        "SubCommand": "fullload",
                        "Flag": " --timeout=12s",
                        "Status": "Success",
                        "Error": "",
                        "CreateTime": "2020-06-28T11:24:49.1627476Z",
                        "UpdateTime": "2020-06-28T11:25:05.7824735Z"
                },
                {
                        "Uid": "21291b73661c421d",
                        "Command": "cpu",
                        "SubCommand": "fullload",
                        "Flag": " --timeout=1m2s",
                        "Status": "Destroyed",
                        "Error": "",
                        "CreateTime": "2020-06-28T11:22:32.9841691Z",
                        "UpdateTime": "2020-06-28T11:23:17.6597329Z"
                },
                {
                        "Uid": "e81105504dfa17f1",
                        "Command": "cpu",
                        "SubCommand": "fullload",
                        "Flag": " --timeout=23",
                        "Status": "Destroyed",
                        "Error": "",
                        "CreateTime": "2020-06-28T11:21:44.4091726Z",
                        "UpdateTime": "2020-06-28T11:22:09.5419411Z"
                }
        ]
}
root@f939a9139919:/app# ./blade s --type c
{
        "code": 200,
        "success": true,
        "result": [
                {
                        "Uid": "41aa902b4a74d201",
                        "Command": "cpu",
                        "SubCommand": "fullload",
                        "Flag": " --timeout=12s",
                        "Status": "Destroyed",
                        "Error": "",
                        "CreateTime": "2020-06-28T11:24:49.1627476Z",
                        "UpdateTime": "2020-06-28T11:25:34.3060006Z"
                },
                {
                        "Uid": "21291b73661c421d",
                        "Command": "cpu",
                        "SubCommand": "fullload",
                        "Flag": " --timeout=1m2s",
                        "Status": "Destroyed",
                        "Error": "",
                        "CreateTime": "2020-06-28T11:22:32.9841691Z",
                        "UpdateTime": "2020-06-28T11:23:17.6597329Z"
                },
                {
                        "Uid": "e81105504dfa17f1",
                        "Command": "cpu",
                        "SubCommand": "fullload",
                        "Flag": " --timeout=23",
                        "Status": "Destroyed",
                        "Error": "",
                        "CreateTime": "2020-06-28T11:21:44.4091726Z",
                        "UpdateTime": "2020-06-28T11:22:09.5419411Z"
                }
        ]
}
```
</details>

File: cli/cmd/create.go [hunk 1]
```diff
@@ -93,15 +90,20 @@ func (cc *CreateCommand) actionRunEFunc(target, scope string, actionCommand *act
...
+			_, errTimeDuartion := time.ParseDuration(tt)
+
+			//error handling
+			if errNumber != nil && errTimeDuartion != nil {
+				if errTimeDuartion != nil {
+					return errTimeDuartion
 				}
+				return errNumber
+
 			}
```
[the remark this answers is not in the document]
Comment (Swarag-N) [reply cli/cmd/create.go]: Yes, I have changed that,

**Pull request merged**